census-us/custom-recipes/address-batch-geocoder-us-census-geo/recipe.py

```diff
@@ -0,0 +1,208 @@
+# -*- coding: utf-8 -*-
+import dataiku
+import pandas as pd, numpy as np
+from dataiku import pandasutils as pdu
+import requests 
+#import time
+from dataiku.customrecipe import *
+import sys
+import re
+import geocoder_utils
+import common
+import os
+
+
+logging.info('1/6 Creating base folder...' )
+path_datadir_tmp = dataiku.get_custom_variables()["dip.home"] + '/tmp/'
+P_CENSUS_CONTENT = 'geocoder'
+FOLDER_NAME = 'tmp_census_us_'+ P_CENSUS_CONTENT 
+common.create_folder(path_datadir_tmp,FOLDER_NAME,True)  
+
+input_name = get_input_names_for_role('input')[0]
+
+output_ = get_output_names_for_role('output')[0]
+output_dataset = dataiku.Dataset(output_)
+
+P_COL_STREET = get_recipe_config()['p_col_street']
+P_COL_CITY = get_recipe_config()['p_col_city']
+P_COL_STATE = get_recipe_config()['p_col_state']
+P_COL_ZIPCODE = get_recipe_config()['p_col_zipcode']
+
+P_BENCHMARK = get_recipe_config()['p_benchmark']
+P_VINTAGE = get_recipe_config()['p_vintage']
+
+
+if P_BENCHMARK=="9":
+    P_VINTAGE ="910"
+
+logging.info('2/6 Parameters:')
+logging.info('[+] BENCHMARK = {} ; VINTAGE = {} '.format(P_BENCHMARK,P_VINTAGE))
+    
+P_BATCH_SIZE_UNIT = int(get_recipe_config()['param_batch_size'])
+if P_BATCH_SIZE_UNIT is None:
+    P_BATCH_SIZE_UNIT = 5000
+    
+
+id_column = get_recipe_config()['p_col_id_column'] 
+id_as_int = get_recipe_config()['param_id_as_int'] 
+
+P_KEEP_NON_MATCHING = get_recipe_config()['param_keep_non_matching']
+#P_RETRY_TIE  = True #get_recipe_config()['param_retry_tie'] ### Potential optimization by resubmitting the ties.
+
+in_cols = [id_column,P_COL_STREET,P_COL_CITY,P_COL_STATE,P_COL_ZIPCODE]
+    
+if id_as_int:
```

Comment:
can't we do some sort of duck typing here rather than toggling and specifically casting the variable? Seems more idiomatic